test/inductor/test_torchinductor_dynamic_shapes.py

```diff
@@ -155,6 +157,119 @@ def fn(x, a):
         ref = fn(x, x.size(0))
         self.assertEqual(res, ref)
 
+    # not supported yet on cpu, https://github.com/pytorch/pytorch/issues/109897
+    @onlyCUDA
+    @torch._dynamo.config.patch(capture_dynamic_output_shape_ops=True)
+    def test_bool_mask_nobreak(self, device):
+        def f(x, b):
+            return (x[b] * 2).sum()
+
+        opt_f = torch.compile(f, fullgraph=True)
+        x = torch.randn(5, device=device)
+        b = torch.tensor([True, True, False, False, True], device=device)
+        r = f(x, b)
+        opt_r = opt_f(x, b)
+        self.assertEqual(r, opt_r)
+
+    @onlyCUDA
+    @torch._dynamo.config.patch(capture_dynamic_output_shape_ops=True)
+    def test_nonzero_size_factory_nobreak(self, device):
+        def f(x, b):
+            y = torch.nonzero(b)
+            return x.new_zeros(y.size(0))
+
+        opt_f = torch.compile(f, fullgraph=True)
+        x = torch.randn(5, device=device)
+        b = torch.tensor([True, True, False, False, True], device=device)
+        r = f(x, b)
+        opt_r = opt_f(x, b)
+        self.assertEqual(r, opt_r)
+
+    @torch._dynamo.config.patch(capture_scalar_outputs=True)
+    def test_item_nobreak(self, device):
+        @torch.compile(fullgraph=True)
+        def f(x):
+            y = x.item()
+            return torch.empty(y)
+
+        f(torch.tensor([3], device=device))
+
+    @onlyCUDA
+    @torch._dynamo.config.patch(capture_scalar_outputs=True)
+    def test_item_zeros_nobreak(self, device):
+        @torch.compile(fullgraph=True)
+        def f(x):
+            y = x.item()
+            torch.empty(y)
+            # This will avoid a NopSchedulerNode
+            return x.new_zeros(y)
+
+        f(torch.tensor([3], device=device))
+
+    @torch._dynamo.config.patch(capture_scalar_outputs=True)
+    @torch._inductor.config.patch(implicit_fallbacks=True)
+    def test_item_to_inputs_kernel_nobreak(self, device):
+        lib = torch.library.Library("test", "DEF")
+
+        try:
+
+            @custom_ops.custom_op("test::foo")
+            def foo(x: torch.Tensor, y: int) -> torch.Tensor:
+                raise NotImplementedError()
+
+            @custom_ops.impl("test::foo")
+            def foo_impl(x: torch.Tensor, y: int) -> torch.Tensor:
+                return x.clone()
+
+            @torch.library.impl_abstract("test::foo", lib=lib)
+            def foo_meta(x: torch.Tensor, y: int) -> torch.Tensor:
+                return x.clone()
+
+            @torch.compile(fullgraph=True)
+            def f(x, r):
+                y = x.item()
+                return torch.ops.test.foo(r, y)
+
+            f(torch.tensor([3], device=device), torch.randn(10, device=device))
+
+        finally:
+            custom_ops._destroy("test::foo")
```

Comment:
@zou3519 take a look at these test cases (here and below) and tell me how I'm doing it wrong :P

Reply:
yeah, essentially the regs here are annoyingly asymmetric but as best I could tell there was no way to actually do it symmetrically